src/register/mstatus.rs

```diff
@@ -57,6 +57,154 @@ pub enum SPP {
 }
 
 impl Mstatus {
+    /// User Interrupt Enable
+    #[inline]
+    pub fn set_uie(&mut self, value: bool) {
```

Comment:
I'm very rusty atm (pun not intended) not being working on these things for some time, but shouldn't these setters be `unsafe` to not hide the fact?

Reply:
emmm, these does not actually change the `mstatus` register, these just change the `bits` field in the `Mstatus` structure.
I do not think an `unsafe` is necessary.

I'm not sure if I followed you correctly, could you please elaborate?

Reply:
This is for the use case like, you want to store `mstatus` in a context, but you do not want to actually change the `mstatus` register.

So you can set up `Mstatus` struct and use asmebly or perhaps a new function to write the `Mstatus` into the `mstatus` register later.

Reply:
Ah I see, your original statement `But I do concern about the amibigiouty of these instance method comparing to the register setters.` makes perfect sense now :D I guess I fell into that trap nicely.

Reply:
XD, So do you think this is a good add-on?
I'm not so sure, but I do feel this useful and makes my code more clear when writing code.
After all, without this, you can only write something like 
```
let later_mstatus = flag_A | flag_B;
/* a bunch of code */
asm!("
    csrw mstatus, a0
    mret
")
```
which is tedious, and too much like C.
Perhaps we need a better naming for this?

Reply:
I think we need to de-confuse this somehow.

Right now the `mstatus::set_mpp(MPP)` and your `mstatus::set_mpp(&mut self, MPP)` are just too confusing (just to provide an example).

Maybe the methods that don't apply should be called something else, more obvious to keep things clear. I'm not quite sure about the naming tho. Perhaps `bitset` instead of `set` prefix?

Reply:
I think `bitset` is pretty good, as it is similar to the `bit_field` naming.

I also added a new method called `apply` to apply the changes in `self.bits` to the actual `mstatus` register and added a bunch of comments in the latest commit.

If this is ok, I'd like to add this `bitset` for other registers, like `mideleg` and `medeleg`.

Reply:
@Disasm any strong opinions? I'm inclined to think this is good now